src/arrayMethodJoin.js

```diff
@@ -4,8 +4,30 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = '';
+
+    let workingSeparator = separator;
+
+    if (workingSeparator === null) {
+      workingSeparator = 'null';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      let element = this[i];
+
+      if (element === null || element === undefined) {
+        element = '';
+      }
+
+      if (i !== this.length - 1) {
+        string += element + workingSeparator;
+      } else {
+        string += element;
+      }
```

Comment:
In this `if` statement, check if this is the last item in the array. If not, then add a separator to the string.
You only need to add a separator here. We will expand `string` above.